### PR TITLE
LINK-1988 | REST API for data analytics purposes

### PR DESCRIPTION
**Pull request by harriris-vincit**
### Description
Adds a simple read-only REST API to allow integrating Linked Events with a data analytics system like Microsoft Power BI. The API supports filtering some of the endpoints using their objects' modification timestamps - this depends on the model if it supports such.

Since the API is dedicated for a specific purpose, it uses it's own token authentication so that users of the normal API cannot authenticate against it and vice versa. This gets rid of the need to consider the usual user permissions for accessing signups, for example. The tokens will automatically expire in 30 days if they are not used (using extends the expiry time automatically).
### Closes
[LINK-1988](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1988)

[LINK-1988]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1988?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr914.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr914.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr914.api.dev.hel.ninja** :rocket::rocket::rocket:

**Review by tuomas777 — APPROVED** [review]
LGTM, great job! 👍 🥇

File: data_analytics/admin.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+from django.contrib import admin
+from django.utils import timezone
+from knox import crypto
+from knox.models import AuthToken
+from knox.settings import CONSTANTS, knox_settings
+
+from data_analytics.forms import DataAnalyticsAuthTokenAdminForm
+from data_analytics.models import DataAnalyticsAuthToken
+from helevents.models import User
```
[the remark this answers is not in the document]
Comment (harriris-vincit) [reply data_analytics/admin.py]: Changed as suggested

File: data_analytics/api.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+
+class DataAnalyticsBaseViewSet(AuditLogApiViewMixin, viewsets.ReadOnlyModelViewSet):
+    authentication_classes = [KnoxTokenAuthentication]
+    http_method_names = ["get", "options"]
+    permission_classes = [IsAuthenticated]
+
+
+class KeywordViewSet(DataAnalyticsBaseViewSet):
+    queryset = Keyword.objects.all().order_by(_ORDER_BY_CREATED_TIME)
```
[the remark this answers is not in the document]
Comment (harriris-vincit) [reply data_analytics/api.py]: Removed `all()`s

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr914.api.dev.hel.ninja** :rocket::rocket::rocket:

**Pull request merged**